### PR TITLE
OBDS-135 Add reset db script

### PR DESCRIPTION
**Pull request by hdsldv**

### CODE REVIEW

File: reset-database.sh [hunk 1]
```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env bash
+
+set -e
+
+while :; do
+  echo -n "Provide openbox db name: "
```
Comment (jmiranda) [inline reset-database.sh:6]: Provide database name:

File: reset-database.sh [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+mysql -e "${SQL}"
+echo "Database recreated"
+
+# Grant privileges
+echo "Do you want grant all privileges to app db user?"
+echo -n "y/N: "
+read ANSWER
+
+if [[ ${ANSWER} == "y" || ${ANSWER} == "Y" ]]; then
+  echo -n "Login: "
```
Comment (jmiranda) [inline reset-database.sh:42]: User:

File: reset-database.sh [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  read LOGIN
+  echo -n "Host: "
```
Comment (jmiranda) [inline reset-database.sh:44]: Is there a way for us to default to localhost?

Comment (jmiranda) [reply reset-database.sh:44]: If so, default User should be openboxes. Default Host should be localhost.

File: reset-database.sh [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+  if mysql -Nse 'SHOW DATABASES;' | grep "${DATABASE_NAME}" -q; then
+    break
+  else
+    echo "Database ${DATABASE_NAME} does not exist. Try again"
+  fi
+done
+
+# User confirmation
+echo "Are you sure you want to delete data from ${DATABASE_NAME}?"
```
Comment (jmiranda) [inline reset-database.sh:17]: We should probably describe the full extent to which we will be destroying their data. And then ask them if they are sure.

**Review by jmiranda — CHANGES_REQUESTED** [review]

File: reset-database.sh (hunk 4 above)
Comment (hdsldv) [reply reset-database.sh:17]: I added line, that database will be deleted, not sure if it's enough

File: reset-database.sh (hunk 3 above)
Comment (hdsldv) [reply reset-database.sh:44]: done

File: reset-database.sh (hunk 2 above)
Comment (hdsldv) [reply reset-database.sh:42]: done

File: reset-database.sh (hunk 1 above)
Comment (hdsldv) [reply reset-database.sh:6]: done

**Review by awalkowiak — APPROVED** [review]
@jmiranda I think it looks good. We can merge it, and we can think of how/where to show it on the frontend to the user.

**Review by jmiranda — APPROVED** [review]

File: grails-app/conf/UrlMappings.groovy [hunk 5]
```diff
@@ -626,6 +626,11 @@ class UrlMappings {
             action = [GET: "getSupportLinks"]
         }
 
+        "/api/resettingInstanceLinks"(parseRequest: true) {
```
Comment (awalkowiak) [inline grails-app/conf/UrlMappings.groovy:629]: Change it into `"/api/resettingInstance/command"`

File: grails-app/conf/UrlMappings.groovy [hunk 6]
```diff
@@ -626,6 +626,11 @@ class UrlMappings {
...
+            controller = { "api" }
+            action = [GET: "getResettingInstanceLinks"]
```
Comment (awalkowiak) [inline grails-app/conf/UrlMappings.groovy:631]: `getResettingInstanceCommand`

File: grails-app/controllers/org/pih/warehouse/api/ApiController.groovy [hunk 7]
```diff
@@ -222,4 +222,10 @@ class ApiController {
 
         render([data: supportLinks] as JSON)
     }
+
+    def getResettingInstanceLinks = {
```
Comment (awalkowiak) [inline grails-app/controllers/org/pih/warehouse/api/ApiController.groovy:226]: `getResettingInstanceCommand`

File: src/js/components/reset-instance/ResetInstancePage.jsx [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router-dom';
+
+import { fetchTranslations } from 'actions';
+import apiClient from 'utils/apiClient';
+import Translate from 'utils/Translate';
+
+
+// eslint-disable-next-line no-shadow
+const ResetInstancePage = ({ locale, history, fetchTranslations }) => {
```
Comment (awalkowiak) [inline src/js/components/reset-instance/ResetInstancePage.jsx:13]: Call it `ResettingInstanceInfoPage`

File: grails-app/conf/Config.groovy [hunk 9]
```diff
@@ -1711,6 +1712,12 @@ openboxes.supportLinks = [
     knowledgeBase: 'https://openboxes.helpscoutdocs.com/',
 ]
 
+// Reset an instance
+
+openboxes.resettingInstance = [
```
Comment (awalkowiak) [inline grails-app/conf/Config.groovy:1717]: Change it to:
`openboxes.resettingInstance.command = "wget https://raw.githubusercontent.com/openboxes/openboxes/develop/reset-database.sh | sh"`

File: grails-app/controllers/org/pih/warehouse/api/ApiController.groovy [hunk 10]
```diff
@@ -222,4 +222,10 @@ class ApiController {
...
+        def resettingInstanceLinks = grailsApplication.config.openboxes.resettingInstance
```
Comment (awalkowiak) [inline grails-app/controllers/org/pih/warehouse/api/ApiController.groovy:227]: def resettingInstanceCommand = grailsApplication.config.openboxes.resettingInstance.command

File: grails-app/controllers/org/pih/warehouse/core/ResetInstanceController.groovy [hunk 11]
```diff
@@ -0,0 +1,17 @@
...
+ * The use and distribution terms for this software are covered by the
+ * Eclipse Public License 1.0 (http://opensource.org/licenses/eclipse-1.0.php)
+ * which can be found in the file epl-v10.html at the root of this distribution.
+ * By using this software in any fashion, you are agreeing to be bound by
+ * the terms of this license.
+ * You must not remove this notice, or any other, from this software.
+ **/
+package org.pih.warehouse.core
+
+class ResetInstanceController {
```
Comment (awalkowiak) [inline grails-app/controllers/org/pih/warehouse/core/ResetInstanceController.groovy:12]: Call it `ResettingInstanceInfoController`

File: src/js/components/reset-instance/ResetInstancePage.jsx [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+  }, []);
+
+  return (
+    <div className="modal-page">
+      <div className="modal-page__content position-relative d-flex" style={{ minHeight: '200px' }}>
+        <h3 className="font-weight-bold my-3">
+          <Translate id="react.resetInstance.header.label" defaultMessage="Reset your instance" />
+        </h3>
+        <p>To reset your instance request your system admin to run this script in terminal:</p>
+        <p className="font-weight-bold">wget {resettingInstanceLinks.scriptUrl} | sh</p>
```
Comment (awalkowiak) [inline src/js/components/reset-instance/ResetInstancePage.jsx:33]: Display entire command pulled from api.

File: src/js/components/reset-instance/ResetInstancePage.jsx [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+    });
+  }, []);
+
+  return (
+    <div className="modal-page">
+      <div className="modal-page__content position-relative d-flex" style={{ minHeight: '200px' }}>
+        <h3 className="font-weight-bold my-3">
+          <Translate id="react.resetInstance.header.label" defaultMessage="Reset your instance" />
+        </h3>
+        <p>To reset your instance request your system admin to run this script in terminal:</p>
```
Comment (awalkowiak) [inline src/js/components/reset-instance/ResetInstancePage.jsx:32]: Missing translation

**Review by awalkowiak — CHANGES_REQUESTED** [review]
@kchelstowski I have few minor request changes on the frontend side

**Review by awalkowiak — APPROVED** [review]

**Pull request merged**